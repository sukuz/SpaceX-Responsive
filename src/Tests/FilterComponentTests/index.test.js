import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import FilterComponent from "../../components/filterComponent";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("filter component renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<FilterComponent></FilterComponent>, div);
});

it("renders all headers correctly", () => {
  const { getByText } = render(<FilterComponent />);
  const FilterText = getByText(/Filter/);
  const launchHeader = getByText(/Successful Launch/);
  const landHeader = getByText(/Successful Land/);
  expect(FilterText).toBeInTheDocument();
  expect(launchHeader).toBeInTheDocument();
  expect(landHeader).toBeInTheDocument();
});

it("matches snapshot", () => {
  const filterTree = renderer.create(<FilterComponent />).toJSON();
  expect(filterTree).toMatchSnapshot();
});
