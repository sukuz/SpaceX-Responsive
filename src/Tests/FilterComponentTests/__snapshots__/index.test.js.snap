// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="filterContainer"
>
  <nav>
    <div
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Filters
    </div>
    <div>
      <div
        className="filterLabels"
      >
        Launch Year
      </div>
      <br />
      <div
        className="btn-container"
      >
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2006
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2007
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2008
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2009
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2010
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2011
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2012
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2013
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2014
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2015
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2016
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2017
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2018
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2019
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          2020
        </button>
      </div>
    </div>
  </nav>
  <main>
    <div>
      <header>
        <div
          className="filterLabels"
        >
          Successful Launch
        </div>
      </header>
      <div
        className="btn-container"
      >
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          true
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          false
        </button>
      </div>
    </div>
    <div>
      <header>
        <div
          className="filterLabels"
        >
          Successful Land
        </div>
      </header>
      <div
        className="btn-container"
      >
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          true
        </button>
        <button
          className="buttonstyle"
          onClick={[Function]}
        >
          false
        </button>
      </div>
    </div>
  </main>
</div>
`;
